src/events/propchange.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Implement propchange events:
+ * Events that fire when a specific prop changes
+ */
+
+import symbols from "../util/symbols.js";
+
+const { propchange } = symbols.new;
+const { events } = symbols.known;
+
+export const hooks = {
+	first_constructor_static () {
+		let propchangeEvents = Object.entries(this[events])
+			.filter(([name, options]) => options.propchange)
+			.map(([eventName, options]) => [eventName, options.propchange]);
+
+		if (propchangeEvents.length > 0) {
+			// Shortcut for events that fire when a specific prop changes
+			this[propchange] = Object.fromEntries(propchangeEvents);
+
+			for (let eventName in this[propchange]) {
+				let propName = this[propchange][eventName];
+				let prop = this.props.get(propName);
+
+				if (prop) {
+					(prop.eventNames ??= []).push(eventName);
+				}
+				else {
+					throw new TypeError(`No prop named ${propName} in ${this.name}`);
+				}
+			}
+		}
+	},
+
+	first_connected () {
```

Comment:
```suggestion
	connected () {
```

We should fire such events for every connected instance since they might have event handlers.

Reply:
Our color elements fire `onPROPchange` events during initialization. They won't do that if we run the code in this hook once the first element is connected.
For example, we won't see the space picker value in the `<output>` element if we don't run this hook for every element:
```html
<space-picker onspacechange="this.nextElementSibling.textContent = this.value"></space-picker>
<output></output>
```

Reply:
But if we don't rename it, the events fired during initialization are not handled. Hm. I'm a bit confused now.

Reply:
> `first_connected` should run for every instance, but only the first time it's connected, whereas `connectedCallback()` (and thus, the `connected` hook) runs every time.
> 
> E.g. consider the following:
> 
> ```js
> let el = document.createElement("foo-bar");
> document.body.append(el); // connected, first_connected
> el.remove();
> document.body.append(el); // connected
> ```

That makes sense. Now I got it. Thanks!

Reply:
> > Which events fired during initialization?
> 
> `valuechange` and `spacechange`

Another issue is described by the comment below: callacks added to the hook after it ran will not be executed. We *can* make it so that they are immediately executed in that case, but I don't know if that's desirable for every case.